### PR TITLE
🚩: Final Milestone: React Capstone Project

### PR DESCRIPTION
**Pull request by Ruthmy**
# 🚩: Final Milestone: React Capstone Project - `React` `Redux Toolkit`

## Hi there! @Ruthmy here.

In this project I have developed a Weather Metrics App using the Current Conditions API from AccuWeather. The app is called `Weather App` and it is a single page application (SPA) that allows users to check various weather metrics for different locations.
- The app has been deployed to `GitHub Pages` and you can find it [here](https://ruthmy.github.io/weather-app/).
- To get the data for the weather, I have used the [Current Conditions API](https://developer.accuweather.com/accuweather-current-conditions-api/apis) from `AccuWeather`.

---

## **Website Goals:**

The goals of this project are to:

- Create a mobile web application to display weather metrics.
- Implement a user-friendly interface with a list of weather metrics.
- Allow users to filter weather metrics by location.
- Provide detailed information for each weather metric.
- Apply a visually appealing design based on the provided [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)).

---

## **Project Goals:**

- [x] The project is developed as a single-page application (SPA) using React and Redux.
- [x] Data retrieved from the AccuWeather API is stored in the Redux store.
- [x] Utilize React Router for efficient navigation.
- [x] Each page has a unique route within the SPA.
- [x] Implement React components to create a seamless user experience.
- [x] Integrate Redux to manage state and data retrieval from the API.
- [x] Write unit tests have been implemented to ensure the functionality and reliability of the application.
- [x] Deploy the application online using `GitHub Pages`.
- [x] Record a 5-minute video demonstrating the app's features and code highlights.
- [x] Follow the [GitFlow as a workflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) for version control.
- [x] Follow the best practices for [HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
- [x] Documented the work in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

<br>

---

[Presentation Video - JavaScript Capstone](https://drive.google.com/file/d/1-eSDabquIJCbvbQblqBPvSwbDIvaRfIb/view)

---

<br>

### Thank you for taking the time to review this PR ⭐!

<br>

---

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
<br>

### CODE REVIEW

File: src/components/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { NavLink } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import continentsImages from '../assets/bd/continentsImages';
+import regionListArray from '../assets/bd/regionList';
+
+import logo from '../assets/img/logo.svg';
+import '../styles/Home.css';
+
+const Home = () => {
```
Comment (topeogunleye) [inline src/components/Home.jsx:11]: - [ ] As required by the project requirement, the Home page should have a list of categories that could be filtered using given parameters, e.g. category name. 

For Example: You could have a search bar to search for continents by name

**Review by topeogunleye — COMMENTED** [review]
Hi @Ruthmy,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Git flow Not Used ⚠️

- Pull request is to be made from the feature branch into the develop branch. I understand that on the last block, you may have used this flow for your group projects. However because this week you are working solo and not making any code reviews, you can use a normal git flow. Please see the project requirement [link here](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Make%20sure%20that%20you%20used%20correct%20Gitflow)


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Ruthmy** [comment]
Hi @topeogunleye  I really appreciate your time in reviewing my code and the feedback you have given me.

Even so I need to add that, I already implemented [this changes you asked for first](https://prnt.sc/lbMUd73EHnnD), to include a filter or search by categories. So I implemented a search bar.
But almost an hour later you invalidate the review by GitFow.

Anyway, I reread the project requirements and the [GitFlow guide](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md), and yes, my PR is the merge from dev to main.
I took a few minutes to check the profiles of other colleagues, [for example this one](https://github.com/grauJavier/react-capstone-project/pull/6), and they are approved and under the same workflow, several feature branches with merge in dev and at the end with a merge at the end from dev to main.

I would be very grateful if you could please tell me what happened.

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED ✔️ 
Hi @Ruthmy , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Alt Text
Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by topeogunleye** [comment]
> But almost



> Hi @topeogunleye I really appreciate your time in reviewing my code and the feedback you have given me.
> 
> Even so I need to add that, I already implemented [this changes you asked for first](https://prnt.sc/lbMUd73EHnnD), to include a filter or search by categories. So I implemented a search bar. But almost an hour later you invalidate the review by GitFow.
> 
> Anyway, I reread the project requirements and the [GitFlow guide](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md), and yes, my PR is the merge from dev to main. I took a few minutes to check the profiles of other colleagues, [for example this one](https://github.com/grauJavier/react-capstone-project/pull/6), and they are approved and under the same workflow, several feature branches with merge in dev and at the end with a merge at the end from dev to main.
> 
> I would be very grateful if you could please tell me what happened.

@Ruthmy I asked my Senior Code Reviewers and they suggested that I invalidate. Congratulations on the approval. Congratulations on your Approval 🎊

**Pull request merged**